### ISSUE TITLE
Making requests for API promotion

### ISSUE THREAD

**Issue by ibahloul** [gravitee-io/issues#5526]
# User Story
As an API publisher, I want to make API promotion requests, so that I can create or update API on a different environment from mine

# Further Information
- The API promotion request is a new action accessible in the details of an API
- An API promotion request generates a new task visible only to users on the target environment (% permissions)
- The following elements are created or updated when promoting an API: ~~groups, members,~~ pages, plans, metadata
- There is no permission related to API promotion, only users allowed to update the API definition (scope: API) can promote API
- The list of environments contains all environments of the same organization
- ~~Add a promotion status with a date & the source/target environment~~

# Example

![promote button](https://user-images.githubusercontent.com/82866977/117484124-f2cd5500-af66-11eb-9b26-7fb6dd92b149.PNG)

![image (11)](https://user-images.githubusercontent.com/82866977/123213533-63541500-d4c6-11eb-8424-93f2baedf5a9.png)

# Tests
## Configuration
- [x] Given an APIM installation registered with Cockpit, when I browse the details of an API, then the button "PROMOTE" is enabled
- [x] Given an APIM installation not registered with Cockpit, when I click on the "PROMOTE" button, then a message promoting Cockpit is displayed

## User Permissions
- [x] Given a user allowed to update the definition on an API, when I browse the details of an API, then the "PROMOTE" button is enabled
- [x] Given a user not allowed to update the definition of on API, when I browse the details of an API, then the "PROMOTE" button is disabled

## API Statuts
- [x] Given an API in a review process, when I browse the details of an API, then the "PROMOTE" button is disabled
- [x] Given an API not in a review process, when I browse the details of an API, then the "PROMOTE" button is enabled
- [x] Given a deprecated API, when I browse the details of an API, then the "PROMOTE" button is disabled
- [x] Given a non-deprecated API, when I browse the details of an API, then the "PROMOTE" button is enabled

## Environment List

![image](https://user-images.githubusercontent.com/4112568/127497766-d11dd81b-5622-478c-bd1c-d919df5c8e6b.png)

- [x] Given the cockpit configuration shown in the screenshot above, when I'm logged in on the "Demo" environment, then the environments available for API promotion are: Dev, QA and Prod

## Environment Selection
- [x] Given an organisation with only 1 environment, when I click on the "PROMOTE" button, then the following message is displayed: "No environment is available for API promotion"
- [x] Given an organisation with 2 environments, when I click on the "PROMOTE" button, then the only available environment is pre-selected
- [x] Given an organisation with 3+ environments, when I click on the "PROMOTE" button, then the list of environments contains all the environments of the organisation except the source environment

## API Promotion requesting
- [x] Given an API eligible for API promotion, when I click on the "PROMOTE" button and cancel the action, then no API promotion requests are visible in the DB
- [x] Given an API eligible for API promotion, when I click on the "PROMOTE" button and I select the default environment for an installation, then a new API promotion request is visible in the DB, and the promotion status = `TO_BE_VALIDATED`
- [x] Given an API eligible for API promotion, when I click on the "PROMOTE" button and I select an environment of an installation (different from the default one), then a new API promotion request is visible in the DB, and the promotion status = `TO_BE_VALIDATED`

**Comment by gaetanmaisse** [comment]
Blocked by https://github.com/gravitee-io/gravitee-cockpit/issues/719

**Comment by gaetanmaisse** [comment]
⚠️ Check slack channel to know how to access nightly environment database for testing purposes.
